src/compile/mark/encode/position-range.ts

```diff
@@ -88,7 +89,18 @@ function pointPosition2OrSize(
 
   if (!channelDef && (channel === 'x2' || channel === 'y2') && (encoding.latitude || encoding.longitude)) {
     // use geopoint output if there are lat2/long2 and there is no point position2 overriding lat2/long2.
-    return {[vgChannel]: {field: model.getName(channel)}};
+    const vgSizeChannel = getSizeChannel(channel);
+
+    const size = model.markDef[vgSizeChannel];
+    if (size != null) {
+      return {
+        [vgSizeChannel]: {value: size}
+      };
+    } else {
+      return {
```

Comment:
This case not covered by test. 

It's probably useful to update the comment since the old comment above is outdated (now there are two subcases)

> // use geopoint output if there are lat2/long2 and there is no point position2 overriding lat2/long2.

Reply:
Done